plugins/modules/pacemaker_resource.py

```diff
@@ -0,0 +1,238 @@
+            )),
+            resource_meta=dict(type='list', elements='str'),
+            resource_argument=dict(type='dict', options=dict(
+                argument_action=dict(type='str', choices=[
+                                   'clone', 'master', 'group', 'promotable']),
+                argument_option=dict(type='list', elements='str'),
+            )),
+            disabled=dict(type='bool', default=False),
+            wait=dict(type='int', default=300),
+        ),
+        required_if=[('state', 'present', ['resource_type', 'resource_option'])],
+        supports_check_mode=True
+    )
+    use_old_vardict = False
+    default_state = "present"
+
+    def __init_module__(self):
+        self.runner = pacemaker_runner(self.module)
+        self.does_not = "Error: resource or tag id '{0}' not found".format(
+            self.vars.name)
+        self.error_messages = ["Error: '{0}' already exists".format(self.vars.name), "Error: Resource '{0} does not exist".format(self.vars.name)]
+        self.vars.set('previous_value', self._get())
+        self.vars.set('_value', self.vars.previous_value, output=False, change=True)
+        self.vars.set_meta('value', initial_value=self.vars.previous_value)
+
+    def _process_command_output(self, fail_on_err, ignore_err_msg=""):
+        def process(rc, out, err):
+          if fail_on_err and rc != 0 and err and ignore_err_msg not in err:
+              self.do_raise(
+                'pcs failed with error (rc={0}): {1}'.format(rc, err))
+          out = out.rstrip()
+          self.vars.value = None if out == "" else out
+          return self.vars.value
+        return process
+
+    def _get(self):
+        return self.runner('state name', output_process=self._process_command_output(False)).run(state='status')
+
+    def state_absent(self):
+        with self.runner('state name', output_process=self._process_command_output(True, "does not exist"), check_mode_skip=True) as ctx:
+            ctx.run()
+            self.vars.stdout = ctx.results_out
+            self.vars.stderr = ctx.results_err
+            self.vars.cmd = ctx.cmd
+        self.vars.set('new_value', None, fact=True)
+        self.vars._value = None
+
+    def state_present(self):
+        with self.runner(
+                'state name resource_type resource_option resource_operation resource_meta resource_argument disabled wait', output_process=self._process_command_output(True, "already exists"),
+                check_mode_skip=True) as ctx:
+            ctx.run(
+                state=self.vars.state,
+                name=self.vars.name,
+                resource_type=self.vars.resource_type,
+                resource_option=self.vars.resource_option,
+                resource_operation=self.vars.resource_operation,
+                resource_meta=self.vars.resource_meta,
+                resource_argument=self.vars.resource_argument,
+                disabled=self.vars.disabled,
+                wait=self.vars.wait)
```

Comment:
and since they all seem to be module params, you can also do:
```suggestion
            ctx.run()
```